examples/Examples/ByteString.hs

```diff
@@ -34,5 +37,9 @@ exampleByteStringAdd ::
   forall n c. (KnownNat n, Symbolic c) => ByteString n c -> ByteString n c -> ByteString n c
 exampleByteStringAdd x y = from (from x + from y :: UInt n Auto c)
 
-exampleSHA :: forall n c. SHA2 "SHA256" c n => ByteString n c -> ByteString 256 c
+exampleSHA :: forall n c.
+  SHA2 "SHA256" c n
+  => Div (PaddedLength n 512 64) 512 * 512 ~ PaddedLength n 512 64
+  => n <= PaddedLength n 512 64
```

Comment:
This is always true by the definition of `PaddedLength` and should be cooked up inside `sha2`